libutils/mutex.h

```diff
@@ -36,9 +36,9 @@ extern pthread_mutex_t *cft_server_filter;
 #define ThreadLock(m)     __ThreadLock(m, __func__, __FILE__, __LINE__)
```

Comment:
I don't think the `extern` `pthread_mutex_t` variables above this really belong to *libutils*. Maybe we should have something like *libpromises/locks.h*?

Reply:
_libpromises/locks.h_ already exists. Do you want it to be put there instead?

Reply:
Hmm, those are different kinds of locks. Maybe *libpromises/global_mutex.h* then?